src/main/java/seedu/address/logic/commands/PayCommand.java

```diff
@@ -102,16 +102,23 @@ private static Student createEditedStudent(Student studentToEdit, double amountP
     /**
      * Updates the paid amount of the student based on the hours the student paid.
      */
-    private static PaidAmount updatePaidAmount(Student student, double amountPaid) {
+    private static PaidAmount updatePaidAmount(Student student, double amountPaid) throws CommandException {
         assert student != null && amountPaid > 0;
         double updatedPaid = student.getPaidAmount().value + amountPaid;
-        String amount = Double.toString(updatedPaid);
+        String amount = String.format("%.2f", updatedPaid);
+
+        if (!PaidAmount.isValidPaidAmount(amount)) {
+            throw new CommandException(Messages.MESSAGE_LIMIT);
+        }
```

Comment:
Should we ignore this since this is a private method. Ive already added test cases below to test for invalid PaidAmount.